Build with clang fails when NOFORTRAN=0

Hi, 
I am trying to build OpenBlas with clang on Fedrora using:
```
make CC=clang USE_OPENMP=1
```
```
gfortran -O2 -Wall -m64 -fopenmp   -o cblat1 cblat1.o ../libopenblas_haswellp-r0.3.1.dev.a -lm -lpthread -lgfortran -lm -lpthread -lgfortran 
gfortran -O2 -Wall -m64 -fopenmp   -o zblat1 zblat1.o ../libopenblas_haswellp-r0.3.1.dev.a -lm -lpthread -lgfortran -lm -lpthread -lgfortran 
../libopenblas_haswellp-r0.3.1.dev.a(blas_server.o): In function `exec_blas':
blas_server_omp.c:(.text+0x26e): undefined reference to `__kmpc_fork_call'
../libopenblas_haswellp-r0.3.1.dev.a(blas_server.o): In function `.omp_outlined.':
blas_server_omp.c:(.text+0x307): undefined reference to `__kmpc_for_static_init_8'
blas_server_omp.c:(.text+0x599): undefined reference to `__kmpc_for_static_fini'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:141: zblat1] Error 1
make[1]: *** Waiting for unfinished jobs....
../libopenblas_haswellp-r0.3.1.dev.a(blas_server.o): In function `exec_blas':
blas_server_omp.c:(.text+0x26e): undefined reference to `__kmpc_fork_call'
../libopenblas_haswellp-r0.3.1.dev.a(blas_server.o): In function `.omp_outlined.':
blas_server_omp.c:(.text+0x307): undefined reference to `__kmpc_for_static_init_8'
blas_server_omp.c:(.text+0x599): undefined reference to `__kmpc_for_static_fini'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:138: cblat1] Error 1

```
and when I use NOFORTRAN=0:
```
make CC=clang USE_OPENMP=1 NOFORTRAN=0
```

```
clang -c -O2 -DMAX_STACK_ALLOC=2048 -fopenmp -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DNO_AVX512 -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=32 -DMAX_PARALLEL_NUMBER=1 -DASMNAME=ztrtri_LU_parallel -DASMFNAME=ztrtri_LU_parallel_ -DNAME=ztrtri_LU_parallel_ -DCNAME=ztrtri_LU_parallel -DCHAR_NAME=\"ztrtri_LU_parallel_\" -DCHAR_CNAME=\"ztrtri_LU_parallel\" -DNO_AFFINITY -I../.. -DDOUBLE  -DCOMPLEX -DCOMPLEX -DDOUBLE -DUNIT trtri_L_parallel.c -o ztrtri_LU_parallel.o
clang -c -O2 -DMAX_STACK_ALLOC=2048 -fopenmp -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DNO_AVX512 -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=32 -DMAX_PARALLEL_NUMBER=1 -DASMNAME=ztrtri_LN_parallel -DASMFNAME=ztrtri_LN_parallel_ -DNAME=ztrtri_LN_parallel_ -DCNAME=ztrtri_LN_parallel -DCHAR_NAME=\"ztrtri_LN_parallel_\" -DCHAR_CNAME=\"ztrtri_LN_parallel\" -DNO_AFFINITY -I../.. -DDOUBLE  -DCOMPLEX -DCOMPLEX -DDOUBLE -UUNIT trtri_L_parallel.c -o ztrtri_LN_parallel.o
ar  -ru ../../libopenblas_haswellp-r0.3.1.dev.a strtri_UU_single.o strtri_UN_single.o strtri_LU_single.o strtri_LN_single.o strtri_UU_parallel.o strtri_UN_parallel.o strtri_LU_parallel.o strtri_LN_parallel.o dtrtri_UU_single.o dtrtri_UN_single.o dtrtri_LU_single.o dtrtri_LN_single.o dtrtri_UU_parallel.o dtrtri_UN_parallel.o dtrtri_LU_parallel.o dtrtri_LN_parallel.o ctrtri_UU_single.o ctrtri_UN_single.o ctrtri_LU_single.o ctrtri_LN_single.o ctrtri_UU_parallel.o ctrtri_UN_parallel.o ctrtri_LU_parallel.o ctrtri_LN_parallel.o ztrtri_UU_single.o ztrtri_UN_single.o ztrtri_LU_single.o ztrtri_LN_single.o ztrtri_UU_parallel.o ztrtri_UN_parallel.o ztrtri_LU_parallel.o ztrtri_LN_parallel.o
make[2]: Leaving directory '/home/shahrzad/src/OpenBLAS_clang/OpenBLAS/lapack/trtri'
make[1]: Leaving directory '/home/shahrzad/src/OpenBLAS_clang/OpenBLAS/lapack'
make[1]: Entering directory '/home/shahrzad/src/OpenBLAS_clang/OpenBLAS/lapack-netlib'
Makefile:7: make.inc: No such file or directory
make[1]: *** No rule to make target 'make.inc'.  Stop.
make[1]: Leaving directory '/home/shahrzad/src/OpenBLAS_clang/OpenBLAS/lapack-netlib'
make: *** [Makefile:218: netlib] Error 2
```
for openmp, clang and gfortran I 'm using:
```
libomp-6.0.0-2.fc28.x86_64 : OpenMP runtime for clang
gcc-gfortran-8.0.1-0.20.fc28.x86_64 
clang-6.0.0-5.fc28.x86_64 : A C language family front-end for LLVM
```
Am I doing something wrong here?

--- a/Makefile
+++ b/Makefile
@@ -21,6 +21,15 @@ ifeq ($(BUILD_RELAPACK), 1)
 RELA = re_lapack
 endif
 
+ifeq ($(NOFORTRAN), 0)
+undefine NOFORTRAN
+endif
+
+ifeq ($(NO_FORTRAN), 1)
+undefine NO_FORTRAN
+NOFORTRAN=1
+endif
+
 LAPACK_NOOPT := $(filter-out -O0 -O1 -O2 -O3 -Ofast,$(LAPACK_FFLAGS))
 
 SUBDIRS_ALL = $(SUBDIRS) test ctest utest exports benchmark ../laswp ../bench

--- a/Makefile
+++ b/Makefile
@@ -21,13 +21,15 @@ ifeq ($(BUILD_RELAPACK), 1)
 RELA = re_lapack
 endif
 
-ifeq ($(NOFORTRAN), 0)
-undefine NOFORTRAN
-endif
-
 ifeq ($(NO_FORTRAN), 1)
-undefine NO_FORTRAN
-NOFORTRAN=1
+define NOFORTRAN
+1
+endef
+define NO_LAPACK
+1
+endef
+export NOFORTRAN
+export NO_LAPACK
 endif
 
 LAPACK_NOOPT := $(filter-out -O0 -O1 -O2 -O3 -Ofast,$(LAPACK_FFLAGS))
@@ -56,7 +58,7 @@ endif
 endif
 
 	@echo "  C compiler       ... $(C_COMPILER)  (command line : $(CC))"
-ifndef NOFORTRAN
+ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
 	@echo "  Fortran compiler ... $(F_COMPILER)  (command line : $(FC))"
 endif
 ifneq ($(OSNAME), AIX)
@@ -117,7 +119,7 @@ endif
 endif
 
 tests :
-ifndef NOFORTRAN
+ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
 	touch $(LIBNAME)
 ifndef NO_FBLAS
 	$(MAKE) -C test all
@@ -219,7 +221,7 @@ netlib :
 
 else
 netlib : lapack_prebuild
-ifndef NOFORTRAN
+ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
 	@$(MAKE) -C $(NETLIB_LAPACK_DIR) lapacklib
 	@$(MAKE) -C $(NETLIB_LAPACK_DIR) tmglib
 endif
@@ -240,7 +242,7 @@ prof_lapack : lapack_prebuild
 	@$(MAKE) -C $(NETLIB_LAPACK_DIR) lapack_prof
 
 lapack_prebuild :
-ifndef NOFORTRAN
+ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
 	-@echo "FORTRAN     = $(FC)" > $(NETLIB_LAPACK_DIR)/make.inc
 	-@echo "OPTS        = $(LAPACK_FFLAGS)" >> $(NETLIB_LAPACK_DIR)/make.inc
 	-@echo "POPTS       = $(LAPACK_FPFLAGS)" >> $(NETLIB_LAPACK_DIR)/make.inc
@@ -283,21 +285,21 @@ endif
 endif
 
 large.tgz :
-ifndef NOFORTRAN
+ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
 	if [ ! -a $< ]; then
 	-wget http://www.netlib.org/lapack/timing/large.tgz;
 	fi
 endif
 
 timing.tgz :
-ifndef NOFORTRAN
+ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
 	if [ ! -a $< ]; then
 	-wget http://www.netlib.org/lapack/timing/timing.tgz;
 	fi
 endif
 
 lapack-timing : large.tgz timing.tgz
-ifndef NOFORTRAN
+ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
 	(cd $(NETLIB_LAPACK_DIR); $(TAR) zxf ../timing.tgz TIMING)
 	(cd $(NETLIB_LAPACK_DIR)/TIMING; $(TAR) zxf ../../large.tgz )
 	$(MAKE) -C $(NETLIB_LAPACK_DIR)/TIMING

--- a/Makefile
+++ b/Makefile
@@ -58,7 +58,7 @@ endif
 endif
 
 	@echo "  C compiler       ... $(C_COMPILER)  (command line : $(CC))"
-ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
+ifneq ($(NOFORTRAN), $(filter-out $(NOFORTRAN), 1 2))
 	@echo "  Fortran compiler ... $(F_COMPILER)  (command line : $(FC))"
 endif
 ifneq ($(OSNAME), AIX)
@@ -119,7 +119,7 @@ endif
 endif
 
 tests :
-ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
+ifneq ($(NOFORTRAN), $(filter-out $(NOFORTRAN), 1 2))
 	touch $(LIBNAME)
 ifndef NO_FBLAS
 	$(MAKE) -C test all
@@ -221,7 +221,7 @@ netlib :
 
 else
 netlib : lapack_prebuild
-ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
+ifneq ($(NOFORTRAN), $(filter-out $(NOFORTRAN), 1 2))
 	@$(MAKE) -C $(NETLIB_LAPACK_DIR) lapacklib
 	@$(MAKE) -C $(NETLIB_LAPACK_DIR) tmglib
 endif
@@ -242,7 +242,10 @@ prof_lapack : lapack_prebuild
 	@$(MAKE) -C $(NETLIB_LAPACK_DIR) lapack_prof
 
 lapack_prebuild :
-ifneq ($(NOFORTRAN), $(filter $(NOFORTRAN), 1 2))
+	$(info filter value of NOFORTRAN is:)
+	$(info x$(filter-out $(NOFORTRAN), 1 2)x)
+
+ifneq ($(NOFORTRAN), $(filter-out $(NOFORTRAN), 1 2))
 	-@echo "FORTRAN     = $(FC)" > $(NETLIB_LAPACK_DIR)/make.inc
 	-@echo "OPTS        = $(LAPACK_FFLAGS)" >> $(NETLIB_LAPACK_DIR)/make.inc
 	-@echo "POPTS       = $(LAPACK_FPFLAGS)" >> $(NETLIB_LAPACK_DIR)/make.inc

--- a/ctest/Makefile
+++ b/ctest/Makefile
@@ -102,7 +102,13 @@ clean ::
 	rm -f x*
 
 FLDFLAGS = $(FFLAGS:-fPIC=) $(LDFLAGS)
-CEXTRALIB =
+ifeq ($(USE_OPENMP), 1)
+ifeq ($(F_COMPILER), GFORTRAN)
+ifeq ($(C_COMPILER), CLANG)
+CEXTRALIB = -lomp
+endif
+endif
+endif
 
 # Single real
 xscblat1: $(stestl1o) c_sblat1.o $(TOPDIR)/$(LIBNAME)

--- a/test/Makefile
+++ b/test/Makefile
@@ -122,8 +122,13 @@ endif
 
 
 FLDFLAGS = $(FFLAGS:-fPIC=) $(LDFLAGS)
-CEXTRALIB =
-
+ifeq ($(USE_OPENMP), 1)
+ifeq ($(F_COMPILER), GFORTRAN)
+ifeq ($(C_COMPILER), CLANG)
+CEXTRALIB = -lomp
+endif
+endif
+endif
 
 sblat1 : sblat1.$(SUFFIX) ../$(LIBNAME)
 	$(FC) $(FLDFLAGS) -o sblat1 sblat1.$(SUFFIX) ../$(LIBNAME) $(EXTRALIB) $(CEXTRALIB)

